### ISSUE TITLE
Some builds fail with "FAILED_PRECONDITION: Missing 1 digests"

### ISSUE THREAD

**Issue by ulfjack** [EngFlow/free#6]
Running a simple Python unit test fails with this error:
```
java.io.IOException: io.grpc.StatusRuntimeException: FAILED_PRECONDITION: Missing 1 digests
```

**Comment by ulfjack** [comment]
This is due to incorrect handling of the empty file. According to the spec, the empty file always has to be present in the CAS. This didn't show up in our integration tests because the server was storing the empty file in the local CAS as soon as it is generated, and it is then present for any subsequent actions.

**Comment by anfelbar** [comment]
> This is due to incorrect handling of the empty file. According to the spec, the empty file always has to be present in the CAS. This didn't show up in our integration tests because the server was storing the empty file in the local CAS as soon as it is generated, and it is then present for any subsequent actions.

Can I send your explanation to the prospect? We will need a new free tier container. The workflow is [disable for now](https://github.com/EngFlow/engflow/pull/16063).

**Comment by ulfjack** [comment]
You can send the explanation or point them here (this issue is public). You can also mention that as soon as they get an empty file onto the server, it'll work, and we'll have the fix in the next release.

I'm aware of the workflow being disabled. :-)

**Comment by ulfjack** [comment]
This is fixed in `2.47.1`.
